Redfish json validator and Service Root class

ffdarkpenguin opened the pull request:
This adds RedfishJsonValidator class and ServiceRoot class, respective tests and docs.
Also app.py was moved from api directory to module root (oneview_redfish_toolkit)
Run.py and tests where updated accordingly. 
Closes #17 
Closes #18

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/hewlettpackard/oneview-redfish-toolkit/29)
<!-- Reviewable:end -->

ricardoas commented:
Reviewed 19 of 250 files at r1.
Review status: 18 of 249 files reviewed at latest revision, 18 unresolved discussions.

---

*[oneview_redfish_toolkit/app.py, line 23 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJffahMPzB9ryeDxHW:-KsJffairNiG0J8e4qbs:b69o10o) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/app.py#L23)):*
> ```Python
> from oneview_redfish_toolkit import util
> 
> cfg = util.load_config('oneview_redfish_toolkit/redfish.ini')
> ```

as i expected load_config returns false (should be None) but we are not testing it here before using it...
the whole application should fail gracefully when this happens, right?

---

*[oneview_redfish_toolkit/app.py, line 26 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJg9WYu9GGUpLcBEme:-KsJg9WYu9GGUpLcBEmf:bd47416) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/app.py#L26)):*
> ```Python
> 
> schemas = dict(cfg.items('schemas'))
> schemas_dict = util.load_schemas(cfg['directories']['schema_dir'], schemas)
> ```

same as above...

---

*[oneview_redfish_toolkit/api/redfish_json_validator.py, line 51 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJgpoGItlVAc9ju4dS:-KsJgpoGItlVAc9ju4dT:byijl00) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/redfish_json_validator.py#L51)):*
> ```Python
>         self.redfish = collections.OrderedDict()
> 
>     def _Validate(self):
> ```

_validate

---

*[oneview_redfish_toolkit/api/redfish_json_validator.py, line 70 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJh4ypetU4tyweqt4y:-KsJh4ypetU4tyweqt4z:b-nzpf9h) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/redfish_json_validator.py#L70)):*
> ```Python
>             raise(e.message)
> 
>     def Serialize(self, pretty=False):
> ```

serialize

---

*[oneview_redfish_toolkit/api/redfish_json_validator.py, line 84 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJhDdiy_NguRmbNtwh:-KsJhDdiy_NguRmbNtwi:bgt66xl) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/redfish_json_validator.py#L84)):*
> ```Python
>         '''
> 
>         self._Validate()
> ```

self._validate()

---

*[oneview_redfish_toolkit/api/service_root.py, line 1 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJicMFDZmgvP5-VxqS:-KsJicMFDZmgvP5-VxqT:bfcm7qf) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/service_root.py#L1)):*
> ```Python
> #!./redfish-venv/bin/python
> ```

not needed.

---

*[oneview_redfish_toolkit/api/service_root.py, line 42 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJhjAHYuew8hEtWRd8:-KsJhjAIGCXFN_wKiYvn:bxqynb) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/service_root.py#L42)):*
> ```Python
>         self.redfish["Name"] = "Root Service"
>         self.redfish["RedfishVersion"] = "1.2.0"
>         self.redfish["UUID"] = "85775665-c110-4b85-8989-e6162170b3ec"
> ```

According to Adriane, hpOneView team added endpoints to recover this info... but they didn't launched a new version yet.
I guess we should leave as it is for now.

---

*[oneview_redfish_toolkit/blueprints/service_root.py, line 28 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJj0YNuLBL1k23ZyS3:-KsJj0YNuLBL1k23ZyS4:b-szjlka) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/blueprints/service_root.py#L28)):*
> ```Python
> @service_root.route('/', methods=["GET"])
> def show():
>     obj = ServiceRoot(current_app.schemas_dict["ServiceRoot"])
> ```

we definitely can improve this, ServiceRoot will always be built using this parameters... any suggestions?
`obj = ServiceRoot()` seems the goal to me, it already knows its schema will be the one with `"ServiceRoot"` key...
What if our Validator knew the loader?

---

*[oneview_redfish_toolkit/blueprints/service_root.py, line 29 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJk4C38YPzP6Y6dx3Q:-KsJk4C38YPzP6Y6dx3R:b-eexxuw) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/blueprints/service_root.py#L29)):*
> ```Python
> def show():
>     obj = ServiceRoot(current_app.schemas_dict["ServiceRoot"])
>     json_str = obj.Serialize(True)
> ```

not sure if we should serialize the pretty version to the end user... 

---

*[oneview_redfish_toolkit/tests/test_blueprint_service_root.py, line 45 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJl06z0Q1XF9J1opZ1:-KsJl06z0Q1XF9J1opZ2:b-4gyj1h) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_blueprint_service_root.py#L45)):*
> ```Python
> 
>         with open(
>             'oneview_redfish_toolkit/tests/mockups/ServiceRoot.mok'
> ```

if this is a pure json file, please do not use this extension (i think it is related to audio files) it is a nightmare to open it for editing.

---

*[oneview_redfish_toolkit/tests/test_blueprint_service_root.py, line 47 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJkt7aZdlwU5_LwzNQ:-KsJkt7aZdlwU5_LwzNR:byz1esm) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_blueprint_service_root.py#L47)):*
> ```Python
>             'oneview_redfish_toolkit/tests/mockups/ServiceRoot.mok'
>         ) as F:
>             mok_json = F.read()
> ```

let's leave uppercase letters to class names and constants, ok?

---

*[oneview_redfish_toolkit/tests/test_redfish_json_validator.py, line 27 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJm2brBJGuiVzH9j2s:-KsJm2brBJGuiVzH9j2t:b-ntygse) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_redfish_json_validator.py#L27)):*
> ```Python
> 
> 
> class TestRedfishJsonValidator(unittest.TestCase):
> ```

shouldn't we be testing validation and serialization functions here?

---

*[oneview_redfish_toolkit/tests/test_redfish_json_validator.py, line 29 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJltEmCPot6DuzWRbn:-KsJltEmCPot6DuzWRbo:bwcfbby) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_redfish_json_validator.py#L29)):*
> ```Python
> class TestRedfishJsonValidator(unittest.TestCase):
> 
>     def test(self):
> ```

not sure how this can go wrong

---

*[oneview_redfish_toolkit/tests/test_util_load_config.py, line 34 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmcme7Wc1bKiLl5Y6:-KsJmcme7Wc1bKiLl5Y7:bmzc18q) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_config.py#L34)):*
> ```Python
>     def test_invalid_config_file(self):
>         # Tests if passing a file that does not exists returns false.
>         self.assertFalse(util.load_config('non-exist.ini'))
> ```

should we test for None instead?

---

*[oneview_redfish_toolkit/tests/test_util_load_config.py, line 85 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmkjUKy9J5yfbBqGV:-KsJmkjUKy9J5yfbBqGW:b-m515r9) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_config.py#L85)):*
> ```Python
>                         self.config_file))
> 
> if __name__ == '__main__':
> ```

we do not need this...

---

*[oneview_redfish_toolkit/tests/test_util_load_schemas.py, line 38 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmtug4XuRB0HD0aSH:-KsJmtug4XuRB0HD0aSI:b-yyw0gz) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_schemas.py#L38)):*
> ```Python
>         self.assertFalse(util.load_schemas('non-exist-schema-dir', schemas))
> 
>     def test_valid_schema_dir_invalide_dict(self):
> ```

nit: invalid

---

*[oneview_redfish_toolkit/tests/test_util_load_schemas.py, line 46 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmwUwin9EsoN4cpb0:-KsJmwUwin9EsoN4cpb1:b2zk1bn) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_schemas.py#L46)):*
> ```Python
>         self.assertFalse(util.load_schemas(self.schema_dir, schemas))
> 
>     def test_valid_schema_dir_valide_dict(self):
> ```

nit: valid_dict

---

*[oneview_redfish_toolkit/tests/test_util_load_schemas.py, line 58 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmp2G-y3a69OHaiYo:-KsJmp2G-y3a69OHaiYp:b-m515r9) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_schemas.py#L58)):*
> ```Python
> 
> 
> if __name__ == '__main__':
> ```

we do not need this...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29)*
<!-- Sent from Reviewable.io -->

danielreed commented on oneview_redfish_toolkit/api/service_root.py:
```diff
@@ -0,0 +1,60 @@
+        self.redfish["Id"] = "RootService"
+        self.redfish["Name"] = "Root Service"
+        self.redfish["RedfishVersion"] = "1.2.0"
+        self.redfish["UUID"] = "85775665-c110-4b85-8989-e6162170b3ec"
```
I might be getting ahead of myself here but I thought UUID was supposed to be populated from this OV REST call: https://{appl}/rest/appliance/nodeinfo/version

llvieira commented on oneview_redfish_toolkit/tests/test_service_root.py:
```diff
@@ -0,0 +1,58 @@
+        obj = ServiceRoot(schemas_dict['ServiceRoot'])
+        self.assertTrue(obj._validate())
+
+    def teste_serialize(self):
```
Change the method name **teste_serialize(self)** to **test_serialize(self)**.

llvieira commented on oneview_redfish_toolkit/tests/test_service_root.py:
```diff
@@ -0,0 +1,58 @@
+# under the License.
+
+"""
+    Tests for redfish_json_validator.py
```
Tests for redfish_json_validator.py? Maybe this comment is not necessary.

llvieira commented on oneview_redfish_toolkit/api/redfish_json_validator.py:
```diff
@@ -0,0 +1,91 @@
+        serialize method.
+        Creates and emmty OrderedDict called redfish which must be populated
+        with the redfish data for validation.
+        The Serrialize method always validates the redfish content before
```
Review some words in the docs, as Serrialize and validatrion.

llvieira commented on oneview_redfish_toolkit/blueprints/service_root.py:
```diff
@@ -0,0 +1,34 @@
+    json_str = obj.serialize(True)
+    response = Response(
+        response=json_str,
+        status=200,
```
We can use status.HTTP_200_OK instead of hardcoded 200. The Flask status comes from the package: from flask_api import status

llvieira commented:
Reviewed 22 of 250 files at r1, 7 of 228 files at r2.
Review status: all files reviewed at latest revision, 23 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29)*
<!-- Sent from Reviewable.io -->

llvieira commented on oneview_redfish_toolkit/blueprints/service_root.py:
```diff
@@ -0,0 +1,35 @@
+
+
+@service_root.route('/', methods=["GET"])
+def show():
```
This method also need documentation. And I guess that **get_service_root()** it's a better name...

ffdarkpenguin commented:
Review status: 25 of 28 files reviewed at latest revision, 24 unresolved discussions.

---

*[oneview_redfish_toolkit/app.py, line 23 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJffahMPzB9ryeDxHW:-KsKrgoilqBgdpNDTguD:b-7ql1td) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/app.py#L23)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

as i expected load_config returns false (should be None) but we are not testing it here before using it...
the whole application should fail gracefully when this happens, right?
</blockquote></details>

fixed

---

*[oneview_redfish_toolkit/app.py, line 26 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJg9WYu9GGUpLcBEme:-KsKrhpPwVmFanPVJRux:b-7ql1td) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/app.py#L26)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

same as above...
</blockquote></details>

fixed

---

*[oneview_redfish_toolkit/api/redfish_json_validator.py, line 51 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJgpoGItlVAc9ju4dS:-KsKrkhZctVsqvOmcO0M:b-7ql1td) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/redfish_json_validator.py#L51)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

_validate
</blockquote></details>

fixed

---

*[oneview_redfish_toolkit/api/redfish_json_validator.py, line 70 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJh4ypetU4tyweqt4y:-KsKrlw4fBblZLD348Pf:b-7ql1td) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/redfish_json_validator.py#L70)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

serialize
</blockquote></details>

fixed

---

*[oneview_redfish_toolkit/api/redfish_json_validator.py, line 84 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJhDdiy_NguRmbNtwh:-KsKrn-4Sgjixsr2PDVM:b-7ql1td) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/redfish_json_validator.py#L84)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

self._validate()
</blockquote></details>

fixed

---

*[oneview_redfish_toolkit/api/redfish_json_validator.py, line 31 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsFJamxBZ6AatpSW0qZ-r2-31:-KsKs6Qzr_DCg1YxaqrU:buy0fse) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/43af434143d20c3edce5a642c2ea63a5982fb7ff/oneview_redfish_toolkit/api/redfish_json_validator.py#L31)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Review some words in the docs, as Serrialize and validatrion for example.
</blockquote></details>

fixed. next time point the lines where the errors are

---

*[oneview_redfish_toolkit/api/service_root.py, line 42 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJhjAHYuew8hEtWRd8:-KsKsBEduZURgiFbRXUO:ba5cp02) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/service_root.py#L42)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

According to Adriane, hpOneView team added endpoints to recover this info... but they didn't launched a new version yet.
I guess we should leave as it is for now.
</blockquote></details>

living for when the release is available

---

*[oneview_redfish_toolkit/api/service_root.py, line 42 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsFJamxBZ6AatpSW0q_-r1-42:-KsKsIgDi_OjppzT-S5X:b-xa8sso) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/service_root.py#L42)):*
<details><summary><i>Previously, danielreed (Daniel Reed) wrote…</i></summary><blockquote>

I might be getting ahead of myself here but I thought UUID was supposed to be populated from this OV REST call: https://{appl}/rest/appliance/nodeinfo/version
</blockquote></details>

waiting for new release

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29)*
<!-- Sent from Reviewable.io -->

ffdarkpenguin commented:
Review status: 25 of 28 files reviewed at latest revision, 24 unresolved discussions.

---

*[oneview_redfish_toolkit/blueprints/service_root.py, line 28 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJj0YNuLBL1k23ZyS3:-KsKswjvm4dRTD_-CUkp:b-sklt9s) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/blueprints/service_root.py#L28)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

we definitely can improve this, ServiceRoot will always be built using this parameters... any suggestions?
`obj = ServiceRoot()` seems the goal to me, it already knows its schema will be the one with `"ServiceRoot"` key...
What if our Validator knew the loader?
</blockquote></details>

you requested to remove depencies from the class. my first version was like this and you requested to remove. so please, decide what you want and how you want. 

---

*[oneview_redfish_toolkit/blueprints/service_root.py, line 29 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJk4C38YPzP6Y6dx3Q:-KsKsaYDVSfBPQ_cwrnO:bd0214h) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/blueprints/service_root.py#L29)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

not sure if we should serialize the pretty version to the end user... 
</blockquote></details>

if you want to see the results in a browser you should. And make it pretty is not a problem for interpreters. so I defend we should

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29)*
<!-- Sent from Reviewable.io -->

ffdarkpenguin commented:
Review status: 25 of 28 files reviewed at latest revision, 24 unresolved discussions.

---

*[oneview_redfish_toolkit/api/service_root.py, line 1 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJicMFDZmgvP5-VxqS:-KsKtcxXZx8NTPnCJ90w:b-hfpv0l) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/api/service_root.py#L1)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

not needed.
</blockquote></details>

removed

---

*[oneview_redfish_toolkit/blueprints/service_root.py, line 31 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJX5E_jfuSb3J46xCW-r2-31:-KsKtpckzs6BN768C_wx:bbhtrml) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/43af434143d20c3edce5a642c2ea63a5982fb7ff/oneview_redfish_toolkit/blueprints/service_root.py#L31)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

We can use status.HTTP_200_OK instead of hardcoded 200. The Flask status comes from the package: from flask_api import status
</blockquote></details>

@ricardo do we HAVE to use this? this status will never change.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29)*
<!-- Sent from Reviewable.io -->

ffdarkpenguin commented:
Review status: 25 of 28 files reviewed at latest revision, 24 unresolved discussions.

---

*[oneview_redfish_toolkit/tests/test_blueprint_service_root.py, line 45 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJl06z0Q1XF9J1opZ1:-KsKud4Qv83Zgol2YEoJ:b-e2xqbq) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_blueprint_service_root.py#L45)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

if this is a pure json file, please do not use this extension (i think it is related to audio files) it is a nightmare to open it for editing.
</blockquote></details>

changed to0 .json

---

*[oneview_redfish_toolkit/tests/test_blueprint_service_root.py, line 47 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJkt7aZdlwU5_LwzNQ:-KsKv2xX1VaCsUFZ1xyp:b-2sptz9) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_blueprint_service_root.py#L47)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

let's leave uppercase letters to class names and constants, ok?
</blockquote></details>

lets build a rule book to avoid this. it´s really annoying fixing this kind of thing with no rule written. fixed

---

*[oneview_redfish_toolkit/tests/test_redfish_json_validator.py, line 27 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJm2brBJGuiVzH9j2s:-KsKw5ZuaZc1CVrXsLO5:b-gk6v79) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_redfish_json_validator.py#L27)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

shouldn't we be testing validation and serialization functions here?
</blockquote></details>

I tested this on service root where there data available. need to test here too? it also will be retested on all class we make. let me know if you want a test here

---

*[oneview_redfish_toolkit/tests/test_redfish_json_validator.py, line 29 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJltEmCPot6DuzWRbn:-KsKw4VaAmbSwDsZdJcR:b-k49bs) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_redfish_json_validator.py#L29)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

not sure how this can go wrong
</blockquote></details>

is this a problem? as we do a lot in the constructor can have a error there.

---

*[oneview_redfish_toolkit/tests/test_service_root.py, line 18 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsFJan5ACrDMimVc6yK-r2-18:-KsKwVIIBWAGluaWPk21:b-rjslyp) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/43af434143d20c3edce5a642c2ea63a5982fb7ff/oneview_redfish_toolkit/tests/test_service_root.py#L18)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Tests for redfish_json_validator.py? Maybe this comment is not necessary.
</blockquote></details>

but does not hurt.

---

*[oneview_redfish_toolkit/tests/test_service_root.py, line 44 at r2](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsFJan5ACrDMimVc6yK-r2-44:-KsKwUG-sMb8oL0ZzVcL:b-7ql1td) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/43af434143d20c3edce5a642c2ea63a5982fb7ff/oneview_redfish_toolkit/tests/test_service_root.py#L44)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

Change the method name **teste_serialize(self)** to **test_serialize(self)**.
</blockquote></details>

fixed

---

*[oneview_redfish_toolkit/tests/test_util_load_config.py, line 34 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmcme7Wc1bKiLl5Y6:-KsKx9GTwVUuc3XUj_ks:bm4fz1t) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_config.py#L34)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

should we test for None instead?
</blockquote></details>

fixed after change the functions

---

*[oneview_redfish_toolkit/tests/test_util_load_config.py, line 85 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmkjUKy9J5yfbBqGV:-KsKxLF0jEOpfs3FHc6G:b2d3r28) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_config.py#L85)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

we do not need this...
</blockquote></details>

if someone while editing the ini delete or change something

---

*[oneview_redfish_toolkit/tests/test_util_load_schemas.py, line 38 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmtug4XuRB0HD0aSH:-KsKxViCiN_ycZfkEQkH:b-7ql1td) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_schemas.py#L38)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

nit: invalid
</blockquote></details>

fixed

---

*[oneview_redfish_toolkit/tests/test_util_load_schemas.py, line 46 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmwUwin9EsoN4cpb0:-KsKxX0CoAZwdJ_kTM3v:b-7ql1td) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_schemas.py#L46)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

nit: valid_dict
</blockquote></details>

fixed

---

*[oneview_redfish_toolkit/tests/test_util_load_schemas.py, line 58 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmp2G-y3a69OHaiYo:-KsKx_QGSIsDQFZb-LH8:b-po74u8) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_schemas.py#L58)):*
<details><summary><i>Previously, ricardoas (Ricardo Araújo) wrote…</i></summary><blockquote>

we do not need this...
</blockquote></details>

hurts?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29)*
<!-- Sent from Reviewable.io -->

ffdarkpenguin commented:
Review status: 19 of 28 files reviewed at latest revision, 24 unresolved discussions.

---

*[oneview_redfish_toolkit/blueprints/service_root.py, line 27 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJX5E_jfuSb3J46xCW-r1-27:-KsKzmtlglgI2FljR1kr:b-7ql1td) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/blueprints/service_root.py#L27)):*
<details><summary><i>Previously, llvieira (Lucas Lima Vieira) wrote…</i></summary><blockquote>

This method also need documentation. And I guess that **get_service_root()** it's a better name...
</blockquote></details>

fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29)*
<!-- Sent from Reviewable.io -->

llvieira commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 251 files at r1, 10 of 10 files at r3.
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-:-KsL0qZEE4RgUm12GTj_:bnfp4nl)*
<!-- Sent from Reviewable.io -->

ricardoas commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 11 of 251 files at r1, 2 of 228 files at r2, 8 of 10 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[oneview_redfish_toolkit/tests/test_blueprint_service_root.py, line 47 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJkt7aZdlwU5_LwzNQ:-KsL2DLgCQ0UA6q-CuCe:b-u8syyq) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_blueprint_service_root.py#L47)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

lets build a rule book to avoid this. it´s really annoying fixing this kind of thing with no rule written. fixed
</blockquote></details>

no need for that: https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables

---

*[oneview_redfish_toolkit/tests/test_util_load_schemas.py, line 58 at r1](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-KsJmp2G-y3a69OHaiYo:-KsL27WsXlMGnQPtz9m2:b-uacxnw) ([raw file](https://github.com/hewlettpackard/oneview-redfish-toolkit/blob/ba4dd15e7f7fb4494cb70b4721660eaac1e0f7ec/oneview_redfish_toolkit/tests/test_util_load_schemas.py#L58)):*
<details><summary><i>Previously, ffdarkpenguin wrote…</i></summary><blockquote>

hurts?
</blockquote></details>

a lot!

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/hewlettpackard/oneview-redfish-toolkit/29#-:-KsL2HynClzd1iaIRtkW:bnfp4nl)*
<!-- Sent from Reviewable.io -->
